app/components/UI/FiatOnRampAggregator/Views/AmountToBuy.js

```
@@ -57,30 +60,64 @@ const AmountToBuy = ({ navigation }) => {
 		selectedRegion,
+		setSelectedRegion,
 		selectedAsset,
 		setSelectedAsset,
-		selectedPaymentMethod,
+		selectedFiatCurrencyId,
+		setSelectedFiatCurrencyId,
 	} = useFiatOnRampSDK();
 
+	const [{ data: countries, isFetching: isFetchingCountries, error: errorCountries }, queryGetCountries] =
+		useSDKMethod({
+			method: 'getCountries',
+			onMount: false,
+		});
+
 	const [{ data: dataTokens, error: errorDataTokens, isFetching: isFetchingDataTokens }] = useSDKMethod(
 		'getCryptoCurrencies',
-		{ countryId: selectedCountry, regionId: selectedRegion },
+		{ countryId: selectedCountry?.id, regionId: selectedRegion?.id },
+		selectedPaymentMethod
+	);
+
+	const [{ data: defaultCurrnecy, error: errorDefaultCurrnecy, isFetching: isFetchingDefaultCurrency }] =
+		useSDKMethod('getDefaultFiatCurrency', { countryId: selectedCountry?.id, regionId: selectedRegion?.id });
+
+	const [{ data: currencies, error: errorCurrencies, isFetching: isFetchingCurrencies }] = useSDKMethod(
+		'getFiatCurrencies',
+		{ countryId: selectedCountry?.id, regionId: selectedRegion?.id },
 		selectedPaymentMethod
 	);
 
+	const currentCurrency = useMemo(() => {
+		// whenever user will switch fiat currnecy, we lookup the new selected currency in the fiat currencies list
+		if (currencies && selectedFiatCurrencyId && selectedFiatCurrencyId !== formatId(selectedCountry?.currency)) {
+			return currencies.find((currency) => currency.id === selectedFiatCurrencyId);
+		}
+
+		if (currencies) {
+			return currencies.find((currency) => currency.id === formatId(selectedCountry?.currency));
+		}
+
+		return defaultCurrnecy;
+	}, [currencies, defaultCurrnecy, selectedFiatCurrencyId, selectedCountry?.currency]);
+
 	const [{ data: currentPaymentMethod, error: errorGetPaymentMethod, isFetching: isFetchingGetPaymentMethod }] =
 		useSDKMethod(
 			'getPaymentMethod',
-			{ countryId: selectedCountry, regionId: selectedRegion },
+			{ countryId: selectedCountry?.id, regionId: selectedRegion?.id },
```

What should happen when this is called with undefined values?

---

If regionId is undefined it should be fine (for countries with no states), if countryId is undefined it should throw an error.

---

Got it, and I guess we catch that and place it in `errorGetPaymentMethod`